core/specfem/periodic_tasks/plot_wavefield.hpp

```diff
@@ -30,6 +30,28 @@ class vtkFloatArray;
 
 #endif // NO_VTK
 
+// Forward declarations of friend functions in global namespace
+#ifndef NO_VTK
+namespace specfem::periodic_tasks {
+class plot_wavefield;
+}
+
+template <specfem::display::format format>
+void initialize(specfem::periodic_tasks::plot_wavefield &plotter,
+                vtkSmartPointer<vtkFloatArray> &scalars);
+
+void initialize_display(specfem::periodic_tasks::plot_wavefield &plotter,
+                        vtkSmartPointer<vtkFloatArray> &scalars);
+
+template <specfem::display::format format>
+void run(specfem::periodic_tasks::plot_wavefield &plotter,
+         vtkSmartPointer<vtkFloatArray> &scalars, const int istep);
+
+void run_render(specfem::periodic_tasks::plot_wavefield &plotter,
+                vtkSmartPointer<vtkFloatArray> &scalars);
```

Comment:
```suggestion
namespace specfem::periodic_tasks::impl {

template <specfem::display::format format>
void initialize(specfem::periodic_tasks::plot_wavefield &plotter,
                vtkSmartPointer<vtkFloatArray> &scalars);

void initialize_display(specfem::periodic_tasks::plot_wavefield &plotter,
                        vtkSmartPointer<vtkFloatArray> &scalars);

template <specfem::display::format format>
void run(specfem::periodic_tasks::plot_wavefield &plotter,
         vtkSmartPointer<vtkFloatArray> &scalars, const int istep);

void run_render(specfem::periodic_tasks::plot_wavefield &plotter,
                vtkSmartPointer<vtkFloatArray> &scalars);

}

```
@Rohit-Kakodkar 
???